lib/surro-gate/proxy.rb

```diff
@@ -111,16 +111,23 @@ def thread_stop
     end
 
     def reactor
-      # Atomically get an array of readable/writable monitors
-      monitors = @mutex.synchronize { @selector.select(0.1) || [] }
+      # Atomically get an array of readable monitors while also polling for writables
+      monitors = @mutex.synchronize do
+        @writer.select(0.1)
+        @reader.select(0.1) || []
+      end
       # Call each transmission proc and collect the results
-      callers = monitors.map { |m| m.value.call }
-      # Sleep for a short time if there was no transmission
-      sleep(0.1) if callers.none? && monitors.any?
+      monitors.map { |m| m.value.call }
```

Comment:
Hmm, this seems slightly different than before (correct me if I am wrong).

**Before:**  `monitors` was the result of calling `.select` on the one `@selector` that existed, and it would return anything, regardless if it was a reader or writer (or an empty array if nothing was returned), then this block would call `.call` on each result, regardless if it was a reader or writer.

**Now:** `monitors` is only either just the readers, or an empty array, and `.call` is never executed on the writers.


Am I wrong?  If not, is this change intentional or okay?